caiman/motion_correction.py

```diff
@@ -1552,13 +1594,50 @@ def register_translation(src_image, target_image, upsample_factor=1,
         raise NotImplementedError("Error: register_translation only supports "
                                   "subpixel registration for 2D images")
 
+    if HAS_CUDA and use_cuda:
+        from skcuda.fft import Plan
+        from skcuda.fft import fft as cudafft
+        from skcuda.fft import ifft as cudaifft
+        try:
+            cudactx
+        except:
+            init_cuda_process()
+
     # assume complex data is already in Fourier space
     if space.lower() == 'fourier':
         src_freq = src_image
         target_freq = target_image
     # real data needs to be fft'd.
     elif space.lower() == 'real':
-        if opencv:
+        if HAS_CUDA and use_cuda:
+            # src_image_cpx = np.array(src_image, dtype=np.complex128, copy=False)
+            # target_image_cpx = np.array(target_image, dtype=np.complex128, copy=False)
+
+            image_gpu = gpuarray.to_gpu(np.stack((src_image, target_image)).astype(np.complex128))
```

Comment:
Longer-term we may want to pull this (and its opencv and other implementations) out into separate functions as this function has gotten fairly long. Not something you'd need to do though.